src/components/movie-list-container/index.jsx

```diff
@@ -1,56 +1,74 @@
-import React, { useState, useEffect } from 'react';
-import './MovieListContainer.css';
-import stubs from './mockedMovies.json';
-import { MovieList } from '../movie-list';
-import { Nav } from '../nav';
-import { SubTitle } from '../sub-title';
-import { Sort } from '../sort';
-import { Filter } from '../filter';
-import { EmptyComponent } from '../empty-component';
-
-export const MovieListContainer = () => {
-  const [genre, setGenre] = useState('all');
-  const [sortParam, setSortParam] = useState('release_date');
-  const [mockedMovies, setMockedMovies] = useState([]);
-
-
-  const filterMovies = (movies) => {
-    if (genre === 'all') return movies;
-    return movies.filter((movie) => movie.genres.includes(genre));
-  }
+import React, { useEffect } from "react";
+import "./MovieListContainer.css";
+import { MovieList } from "../movie-list";
+import { Nav } from "../nav";
+import { SubTitle } from "../sub-title";
+import { Sort } from "../sort";
+import { Filter } from "../filter";
+import { EmptyComponent } from "../empty-component";
+import { fetchMoviesActionCreator } from "../../store/actionCreators/fetchMovies";
+import { connect } from "react-redux";
+import { setFilterActionCreator } from "../../store/actionCreators/filter-movies";
+import { setSortActionCreator } from "../../store/actionCreators/sort-movies";
 
+const MovieListContainerTwo = ({
+  films,
+  fetchMovies,
+  loading,
+  genre,
+  sortParam,
+  setFilter,
+  setSort
+}) => {
   const filterByGenre = (genre) => {
-    setGenre(genre);
-  }
+    setFilter(genre);
+  };
 
-  const applySortParam = (sortParam) => {
-    setSortParam(sortParam);
-  }
+  const applySortParam = (sortParam) => setSort(sortParam);
+
+  useEffect(() => {
+    fetchMovies();
+  }, [genre, sortParam]);
 
-  const sortMovies = (movies) => {
-    let copyOfMovies = movies.slice();
-    if (sortParam === 'release_date') {
-      return copyOfMovies.sort((a, b) => new Date(b[sortParam]) - new Date(a[sortParam]));
-    }
-      return copyOfMovies.sort((a, b) => b[sortParam] > a[sortParam] ? 1 : -1);
+  const movieListMakrUp = () => (
+    <>
+      <SubTitle movieListLength={films.length} />
+      <MovieList movies={films} />
+    </>
+  );
+  if (loading) {
+    return <h1>Loading</h1>;
   }
+  return (
+    <React.Fragment>
+      <Nav>
+        <Filter filterMovies={filterByGenre} />
+        <Sort setSortParam={applySortParam} sortParam={sortParam} />
+      </Nav>
+      {films.length !== 0 ? movieListMakrUp() : <EmptyComponent />}
+    </React.Fragment>
+  );
+};
 
-  useEffect(() => {
-    setMockedMovies(stubs);
-  }, []);
+const mapStateToProps = (state) => {
+  return {
+    films: state.films,
+    loading: state.loading,
+    genre: state.genre,
+    sortParam: state.sortParam,
+    state: state
```

Comment:
In which place of component do you use `state`? 🧐
If it's redundant - delete it.